### PR TITLE
feat: [#504] Route supports configure timeout

### PR DESCRIPTION
**Pull request by KlassnayaAfrodita**
github.com/goravel/goravel/issues/504

add WithContext

## 📑 Description

Closes https://github.com/goravel/goravel/issues/504

@coderabbitai summary

## ✅ Checks

- [ ] Added test cases for my code

### CODE REVIEW

**Review by hwbrzzl — COMMENTED** [review]
Please merge your two PRs, one PR is fine. And add the global configuration logic.

**Review by hwbrzzl — COMMENTED** [review]
Please update this PR according to the gin PR.

**Review by hwbrzzl — COMMENTED** [review]
Sorry, there are a lot of issues, please follow the goravel/gin PR to update this.

1. Update go.mod, upgrade goravel/framework to the latest version;
2. Remove the test case in `context_test.go`, just like what goravel/gin did;
3. Update `route.go::GlobalMiddleware`, just like what goravel/gin did;
4. Update `group_test.go`, the test cases should not pass now;
5. Rename `cors.go` to `middleware_cors.go`, the same as `tls.go`;
6. Bebase your branch;

**Comment by hwbrzzl** [comment]
Thanks for your updates. I'm not sure if this PR is ready to be reviewed again, just notice, a few points are still not updated.

File: go.mod [hunk 1]
```diff
@@ -8,11 +8,11 @@ require (
 	github.com/gofiber/fiber/v2 v2.52.5
 	github.com/gofiber/template/html/v2 v2.1.2
 	github.com/gookit/validate v1.5.2
-	github.com/goravel/framework v1.14.1-0.20240930062942-b310a98799ee
+	github.com/goravel/framework v1.14.7
```
Comment (hwbrzzl) [inline go.mod:11]: Need to run `go get github.com/goravel/framework@master`

File: route_test.go [hunk 2]
```diff
@@ -485,6 +485,29 @@ func TestStop(t *testing.T) {
 	}
 }
 
+func TestTimeoutMiddlewareIntegration(t *testing.T) {
```
Comment (hwbrzzl) [inline route_test.go:488]: This is unnecessary, we can remove it.

**Review by hwbrzzl — COMMENTED** [review]
Great job 👍  You got the key, almost finish.

**Comment by hwbrzzl** [comment]
Unfortunately, CI still failed.

**Comment by hwbrzzl** [comment]
You  can ask me if you have any questions for this.

**Comment by KlassnayaAfrodita** [comment]
> Если у вас возникнут какие-либо вопросы по этому поводу, вы можете задать их мне.

Do I understand correctly that I need to update the mock files?

**Comment by hwbrzzl** [comment]
> > You can ask me if you have any questions for this .
> 
> Do I understand correctly that I need to update the mock files?

No, unit tests failed, did you run unit tests locally?

image

**Comment by hwbrzzl** [comment]
Yes, you need to optimize the test flow, there are some unexpected mockConfig assertion.

**Comment by hwbrzzl** [comment]
Hey @KlassnayaAfrodita Are you still working on this, please? If you feel it's difficult to fix the CI, you can add me to your fiber repo, I can solve it.

**Comment by KlassnayaAfrodita** [comment]
I invited you to the repository, could you then describe how this problem is solved. current branch patch-1

**Review by hwbrzzl — APPROVED** [review]

**Pull request merged**